src/screens/Chat/GroupChatListScreen.js

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from "react";
+import {
+  Alert,
+  Platform,
+  ScrollView,
+  StyleSheet,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  View
+} from "react-native";
+import { useNavigation } from "@react-navigation/native";
+import { chatType } from "../../constants/Chat";
+import {
+  fetchActiveGroupChats,
+  checkUnreadPrivateMessages
+} from "../../services/Chat/FetchActiveChats";
+import ActiveChatLists from "../../components/Chat/ActiveChatLists";
+import NotificationText from "../../components/Miscellaneous/NotificationText";
+
+const GroupChatListScreen = () => {
+  const [search, setSearch] = useState("");
+  const [activeChats, setActiveChats] = useState([]);
+  const [isUnreadExists, setIsUnreadExists] = useState(false);
+  const [expand, setExpand] = useState(null);
+  const navigation = useNavigation();
+
+  const expandStatus = (value) => expand === value;
+
+  const changeExpand = (value) => {
+    if (expandStatus(value)) {
+      setExpand(null);
+    } else {
+      setExpand(value);
+    }
+  };
+
+  useEffect(() => {
+    return fetchActiveGroupChats({
+      onSuccess: (data) => {
+        setActiveChats(data);
+      },
+      onFailure: (error) => {
+        Alert.alert("Error", error.message);
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    return checkUnreadPrivateMessages({
+      onFound: () => {
+        setIsUnreadExists(true);
+      },
+      onNotFound: () => {
+        setIsUnreadExists(false);
+      },
+      onFailure: (error) => {
+        Alert.alert("Error", error.message);
+      }
+    });
+  }, []);
+
+  return (
+    <ScrollView style={styles.container}>
+      <TextInput
+        value={search}
+        onChangeText={setSearch}
+        placeholder="Search messages"
+        style={styles.textInput}
+      />
+
+      <Text style={styles.title}>Group Chat List</Text>
+
+      <View style={styles.buttonGroup}>
+        <TouchableOpacity
+          style={styles.button}
+          onPress={() =>
+            navigation.navigate("Friends", { screen: "GroupList" })
+          }
+        >
+          <Text style={styles.buttonText}>Message other groups</Text>
+        </TouchableOpacity>
+
+        <TouchableOpacity
+          style={styles.button}
+          onPress={() => navigation.navigate("ChatList")}
+        >
```

Comment:
I don't know why but there might be a problem with this navigation (or the stack navigator) where if you press this button and press back (from your phone, not the button) , you will not go back to the group chat list compared to when you go to the private chat list and press the equivalent button and then press back